### PR TITLE
feat(docs): document `opsgenie` package

### PR DESCRIPTION
**Pull request by pierwill**

### CODE REVIEW

File: stdlib/contrib/sranka/opsgenie/opsgenie.flux [hunk 1]
```diff
@@ -1,26 +1,73 @@
...
 import "strings"
 
-// respondersToJSON converts an array of responder strings to JSON array that can be embedded into an alert message
+// respondersToJSON converts an array of Opsgenie responder strings
+// to a string-encoded JSON array that can be embedded in an alert message.
+//
+// ## Parameters
+// - v: (Required) Array of Opsgenie responder strings.
+//   Responder strings must begin with
+//   `user: `, `team: `, `escalation: `, or `schedule: `.
```
Comment (sanderson) [inline stdlib/contrib/sranka/opsgenie/opsgenie.flux:19]:
~~~suggestion
// - v: Array of Opsgenie responder strings.
//
//   Responder strings must begin with
//   `user: `, `team: `, `escalation: `, or `schedule: `.
//
// ## Examples
// ### Convert strings to Opsgenie responder strings
// ```
// import "contrib/sranka/opsgenie"
//
// opsgenie.respondersToJSON(
//     v: [
//         "user: example-user",
//         "team: example-team",
//         "escalation: example-escalation",
//         "schedule: example-schedule",
//     ],
// )
//
// // Returns the following JSON string:
// // [{"type":"user","username":" example-user"},{"type":"team","name":" example-team"},{"type":"escalation","name":" example-escalation"},{"type":"schedule","name":" example-schedule"}]
// ```
~~~

File: stdlib/contrib/sranka/opsgenie/opsgenie.flux [hunk 2]
```diff
@@ -1,26 +1,73 @@
...
-// `actions`     - array  - Array of strings that specifies actions that will be available for the alert. 
-// `details`     - string - Additional details of an alert, it must be a JSON-encoded map of key-value string pairs. 
-// `visibleTo`   - array  - Arrays of teams and users that the alert will become visible to without sending any notification. Optional. 
+// sendAlert sends an alert message to Opsgenie.
+// 
+// ## Parameters
+// 
+// - url: Opsgenie API URL. Defaults to `https://api.opsgenie.com/v2/alerts`.
+// - apiKey: (Required) Opsgenie API authorization key.
+// - message: (Required) Alert message text.
```
Comment (sanderson) [inline stdlib/contrib/sranka/opsgenie/opsgenie.flux:28]:
```suggestion
// - apiKey: Opsgenie API authorization key.
// - message: Alert message text.
```

File: stdlib/contrib/sranka/opsgenie/opsgenie.flux [hunk 3]
```diff
@@ -1,26 +1,73 @@
...
+// - description: Alert description. 15000 characters or less.
+// - priority: Opsgenie alert priority.
+//   Valid values include:
+//   - `P1`
+//   - `P2`
+//   - `P3` (default)
+//   - `P4`
+//   - `P5`
+// - responders: List of responder teams or users.
+//   Use the `user: ` prefix for users and `teams: ` prefix for teams.
```
Comment (sanderson) [inline stdlib/contrib/sranka/opsgenie/opsgenie.flux:42]:
```suggestion
// - priority: Opsgenie alert priority.
//
//   **Valid values**:
//   - `P1`
//   - `P2`
//   - `P3` (default)
//   - `P4`
//   - `P5`
//
// - responders: List of responder teams or users.
//   Use the `user: ` prefix for users and `teams: ` prefix for teams.
```

File: stdlib/contrib/sranka/opsgenie/opsgenie.flux [hunk 4]
```diff
@@ -1,26 +1,73 @@
...
+//     |> range(start: -1m)
+//     |> filter(fn: (r) => r._measurement == "statuses")
+//     |> last()
+//     |> findRecord(fn: (key) => true, idx: 0)
+//     opsgenie.sendAlert(
+//       apiKey: apiKey,
+//       message: "Disk usage is: ${lastReported.status}.",
+//       alias: "example-disk-usage",
+//       responders: ["user:john@example.com", "team:itcrowd"]
+//     )
```
Comment (sanderson) [inline stdlib/contrib/sranka/opsgenie/opsgenie.flux:69]: `flux fmt`'d

```suggestion
// import "influxdata/influxdb/secrets"
// import "contrib/sranka/opsgenie"
// 
// apiKey = secrets.get(key: "OPSGENIE_APIKEY")
//
// lastReported = from(bucket: "example-bucket")
//     |> range(start: -1m)
//     |> filter(fn: (r) => r._measurement == "statuses")
//     |> last()
//     |> findRecord(fn: (key) => true, idx: 0)
//
// opsgenie.sendAlert(
//     apiKey: apiKey,
//     message: "Disk usage is: ${lastReported.status}.",
//     alias: "example-disk-usage",
//     responders: ["user:john@example.com", "team:itcrowd"],
// )
```

File: stdlib/contrib/sranka/opsgenie/opsgenie.flux [hunk 5]
```diff
@@ -1,26 +1,73 @@
...
+// ```
```
Comment (sanderson) [inline stdlib/contrib/sranka/opsgenie/opsgenie.flux:70]:
```suggestion
// ```
//
// tags: single notification
```

File: stdlib/contrib/sranka/opsgenie/opsgenie.flux [hunk 6]
```diff
@@ -60,12 +107,62 @@ sendAlert = (
...
-// `url`         - string - Opsgenie API URL. Defaults to "https://api.opsgenie.com/v2/alerts". 
-// `apiKey`      - string - API Authorization key. 
-// `entity`      - string - Entity of the alert, used to specify domain of the alert. Optional. 
-// The returned factory function accepts a `mapFn` parameter.
-// The `mapFn` must return an object with all properties defined in the `sendAlert` function arguments (except url, apiKey and entity).
+// endpoint sends an alert message to Opsgenie using data from table rows.
+// 
+// ## Parameters
+// - url: Opsgenie API URL. Defaults to `https://api.opsgenie.com/v2/alerts`.
+// - apiKey: (Required) Opsgenie API authorization key.
```
Comment (sanderson) [inline stdlib/contrib/sranka/opsgenie/opsgenie.flux:114]:
```suggestion
// - url: Opsgenie API URL. Default is `https://api.opsgenie.com/v2/alerts`.
// - apiKey: Opsgenie API authorization key.
```

File: stdlib/contrib/sranka/opsgenie/opsgenie.flux [hunk 7]
```diff
@@ -60,12 +107,62 @@ sendAlert = (
...
+// - entity: Alert entity used to specify the alert domain.
+// 
+// ## Usage
+// opsgenie.endpoint is a factory function that outputs another function.
```
Comment (sanderson) [inline stdlib/contrib/sranka/opsgenie/opsgenie.flux:118]:
```suggestion
// `opsgenie.endpoint()` is a factory function that outputs another function.
```

File: stdlib/contrib/sranka/opsgenie/opsgenie.flux [hunk 8]
```diff
@@ -60,12 +107,62 @@ sendAlert = (
...
+// - alias
+// - description
+// - priority
+// - responders
+// - tags
+// - actions
+// - details
+// - visibleTo
+// 
+// For more information, see `opsgenie.sendAlert`.
```
Comment (sanderson) [inline stdlib/contrib/sranka/opsgenie/opsgenie.flux:136]:
```suggestion
// For more information, see `opsgenie.sendAlert()`.
```

File: stdlib/contrib/sranka/opsgenie/opsgenie.flux [hunk 9]
```diff
@@ -60,12 +107,62 @@ sendAlert = (
...
+//       description: "",
+//       priority: "P3",
+//       responders: ["user:john@example.com", "team:itcrowd"],
+//       tags: [],
+//       entity: "my-lab",
+//       actions: [],
+//       details: "{}",
+//       visibleTo: []
+//     })
+//   )()
```
Comment (sanderson) [inline stdlib/contrib/sranka/opsgenie/opsgenie.flux:164]:
```suggestion
// import "influxdata/influxdb/secrets"
// import "contrib/sranka/opsgenie"
// 
// apiKey = secrets.get(key: "OPSGENIE_APIKEY")
// endpoint = opsgenie.endpoint(apiKey: apiKey)
// 
// crit_statuses = from(bucket: "example-bucket")
//     |> range(start: -1m)
//     |> filter(fn: (r) => r._measurement == "statuses" and status == "crit")
// 
// crit_statuses
//     |> endpoint(
//         mapFn: (r) => ({
//             message: "Great Scott!- Disk usage is: ${r.status}.",
//             alias: "disk-usage-${r.status}",
//             description: "",
//             priority: "P3",
//             responders: ["user:john@example.com", "team:itcrowd"],
//             tags: [],
//             entity: "my-lab",
//             actions: [],
//             details: "{}",
//             visibleTo: [],
//         }),
//     )()
```

File: stdlib/contrib/sranka/opsgenie/opsgenie.flux [hunk 10]
```diff
@@ -60,12 +107,62 @@ sendAlert = (
...
+// ```
```
Comment (sanderson) [inline stdlib/contrib/sranka/opsgenie/opsgenie.flux:165]:
```suggestion
// ```
//
// tags: notification endpoints,transformations
```

**Review by sanderson — CHANGES_REQUESTED** [review]

**Review by wolffcm — APPROVED** [review]
Looks good to me.

**Review by sanderson — APPROVED** [review]

**Pull request merged**